apps/judicial-system/backend/src/app/modules/case-table/caseTable.cellGenerators.ts

```diff
@@ -150,6 +154,99 @@ const generateRequestCaseStateTag = (c: Case, user: TUser): TagValue => {
   }
 }
 
+const generateIndictmentCaseStateTag = (c: Case, user: TUser): TagValue => {
+  const {
+    state,
+    indictmentRulingDecision,
+    indictmentDecision,
+    indictmentReviewerId,
+  } = c
+
+  const courtDate = DateLog.courtDate(c.dateLogs)?.date
```

Comment:
Please note that `courtDate` in the old `CaseListEntry` is set to either `arraignmentDate` or `courtDate` depending on wheteher the case has a `indictmentDecision`. Therefore, you need to do something similar here to get the correct tag. Note that you are already doing this for the column `indictmentArraignmentDate`.